src/v/ssx/thread_worker.h

```diff
@@ -64,9 +64,19 @@ class worker_task final : public task_base {
     }
 
     void process(ss::alien::instance& alien, ss::shard_id shard) final {
-        auto f = futurator::invoke(_func);
-        ss::alien::run_on(alien, shard, [this, f{std::move(f)}]() mutable {
-            f.forward_to(std::move(_promise));
+        try {
+            set_value(alien, shard, _func());
+        } catch (...) {
+            set_exception(alien, shard, std::current_exception());
+        };
+    }
+
+    void set_value(
+      ss::alien::instance& alien,
+      ss::shard_id shard,
+      typename futurator::value_type v) {
+        ss::alien::run_on(alien, shard, [this, v{std::move(v)}]() {
+            _promise.set_value(std::move(v));
```

Comment:
unless you mark the continuation mutable this move will be a copy.

Reply:
Good spot. Modified the test to ensure this works.